lib/bolt/executor.rb

```diff
@@ -102,6 +110,15 @@ def shutdown
     # defined by the transport. Yields each batch, along with the corresponding
     # transport, to the block in turn and returns an array of result promises.
     def queue_execute(targets)
+      if @warn_concurrency && targets.length > @concurrency
+        @warn_concurrency = false
+        @logger.warn("The ulimit is low, which may cause file limit issues. We've configured concurrency to "\
+                     "'#{@concurrency}' to mitigate those issues, which may cause Bolt to run slow. "\
+                     "Configure ulimit using 'ulimit -n <limit>' in your shell configuration, or ignore this "\
+                     "warning by configuring Bolt's concurrency. "\
+                     "See https://puppet.com/docs/bolt/latest/bolt_known_issues.html for details.")
```

Comment:
```suggestion
        @logger.warn("The ulimit is low, which may cause file limit issues. Default concurrency has "\
                     "'been set to  #{@concurrency}' to mitigate those issues, which may cause Bolt to run "\
                     "slower than usual. Disable this warning by setting the ulimit using 'ulimit -n <limit>' in  "\
                     "your shell configuration or by configuring Bolt's concurrency. "\
                     "See https://puppet.com/docs/bolt/latest/bolt_known_issues.html for details.")
```

I'm not sure if this reads better?